marlowe-dashboard-client/src/Css.purs

```diff
@@ -99,19 +97,31 @@ inputBaseNoFocus :: Array String
 inputBaseNoFocus = inputBase <> [ "focus:ring-0" ]
 
 input :: Boolean -> Array String
-input valid = inputBaseFocus <> [ "border", "bg-transparent" ] <> if valid then [ "border-black", "focus:border-black" ] else [ "border-red" ]
+input valid =
+  inputBaseFocus
+    <> [ "border-gray" ]
+    <> applyWhen (not valid) [ "border-red" ]
```

Comment:
I think an `if else` expression is better as there are no guarantees that the browser will pick the red border from the grey one if both are present (that has bitten me in the past)

Reply:
ah good point 👍